File: internal/extract/testdata/kotlin/basic.kt
import java.util.List

open class Animal(val name: String) {
    open fun speak() {
        println("hello")
    }
}

class Dog(name: String) : Animal(name) {
    override fun speak() {
        println("woof")
    }
}

interface Groomable {
    fun groom()
}

object Singleton {
    fun doWork() {
        println("working")
    }
}

fun main() {
    val d = Dog("Rex")
    d.speak()
}

File: internal/extract/testdata/scala/basic.scala
import scala.collection.mutable.ListBuffer

class Animal(val name: String) {
  def speak(): Unit = {
    println("hello")
  }
}

class Dog(name: String) extends Animal(name) {
  override def speak(): Unit = {
    println("woof")
  }
}

trait Groomable {
  def groom(): Unit
}

object Singleton {
  def doWork(): Unit = {
    println("working")
  }
}
